### ISSUE TITLE
linter: 'typescript/no-unnecessary-type-assertion' not catching 'as const'

### ISSUE THREAD

**Issue by donaldpipowitch** [oxc-project/oxc#22324]
### What version of Oxlint are you using?

1.61.0

### What command did you run?

oxlint

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```ts
import { defineConfig } from 'oxlint';

export default defineConfig({
  options: {
    typeAware: true,
  },
  plugins: [
    'typescript',
  ],
  rules: {
    'typescript/no-unnecessary-type-assertion': [
      'error',
      {
        checkLiteralConstAssertions: true,
      },
    ],
});
```

### What happened?

It looks like ` 'typescript/no-unnecessary-type-assertion'` doesn't catch unnecessary `as const` calls:

```
type GraphType = 'bar' | 'line';

type GraphConfig = {
  label: string;
  type: GraphType;
};

const graphConfig = {
  a: {
    label: 'Some Title',
    type: 'line',
  },
  b: {
    label: 'Another Title',
    type: 'line' as const, // ! as const is redundant
  },
} satisfies Record<string, GraphConfig>;
```

Also without `as const` `a.type` is `type: 'line'` and not `type: GraphType`:

<img width="394" height="152" alt="Image" src="https://github.com/user-attachments/assets/15fddd8a-be0d-4717-83c6-e462fc8762c2" />

**Comment by camc314** [comment]
looks like typescript-eslint doesn't report this case either: https://typescript-eslint.io/play#ts=5.9.2&fileType=.tsx&code=FAFwngDgpgBA4gJwIYQBYBVKwLwwOQBGSCeMAPvgDYCWAdlHgNzChbzJoDCA9rQGbUA5jFwBvYDBiUkBKJQBcMAM4gEdQc0nhoixCgxZmAX2bAAxrxUxBHVD35CRMcZKSKXkqTLmK8AZW4AW1h0ahBKBgAaCU9tKF8aejxoySMUmAJ3GMlpWQV8AEFabhBUKAQYUPCo7Jg4hLoGGCQlGAtaFUiYAHpumABCZtb2q2pWhCgAEwBXWkmkWhAYtOAjZSQQMYEoVoAlKAsESYAeFTVaQS69Ll4BQQA%2BUxYgA&eslintrc=AQ18ZXTwN4CgQCIBOBXANgUwM7IC54lRkABAFwE8AHPAY1QEsaKBaPTJgOwoHpuAezbpu3bPTy4AhqiptqdNtNy5sqCk0HdCwANokwydakGpkAGkOhEUZPQAWEgNYAZJhXXTMAYW24KAEFVdU1-XQoMbGsQAF9rAF0SeNigA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false
